### PR TITLE
feat: add strict to Field

### PR DESCRIPTION
**Pull request by adriangb**
Chipping away at #4664.

### CODE REVIEW

File: pydantic/fields.py [hunk 1]
```diff
@@ -77,6 +77,9 @@ def __init__(self, **kwargs: Any) -> None:
         self.description = kwargs.get('description')
         self.exclude = kwargs.get('exclude')
         self.include = kwargs.get('include')
+        if 'strict' in kwargs and kwargs['strict'] is None:
+            # not set, don't include in schema
+            kwargs.pop('strict')
```
Comment (samuelcolvin) [inline pydantic/fields.py:82]: This shouldn't be necessary, the schema functions should automatically strip out `None`s.

**Review by samuelcolvin — COMMENTED** [review]
seems like a good start, are you going to add more here?

**Comment by adriangb** [comment]
> are you going to add more here?

Nope! I was planning on _trying_ to make small incremental changes and this seems like a good isolated change that we can merge as is.

**Comment by adriangb** [comment]
@samuelcolvin do you _want_ me to add more here or can we merge this and iterate?

**Review by samuelcolvin — COMMENTED** [review]
Sorry for slow reply, other than that tweak, I think this looks good.

Feel free to merge this and fix this later if it's easier.

File: tests/test_strict.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    from typing_extensions import Annotated
+else:
+    from typing import Annotated
+
+import pytest
+
+from pydantic import BaseModel, Field, ValidationError
+
+
+class ModelWithStrictField(BaseModel):
```
[the remark this answers is not in the document]
Comment (samuelcolvin) [reply tests/test_strict.py:14]: great, lots of examples of how I've done it in `test_main.py`

**Comment by samuelcolvin** [comment]
thanks so much.

**Pull request merged**